pkg/config/draftconfig_template_test.go

```diff
@@ -121,29 +130,47 @@ func loadTemplatesWithValidation() error {
 				return fmt.Errorf("template %s has an invalid variable kind: %s", path, variable.Kind)
 			}
 
-			// version range check once we define versions
-			// if _, err := semver.ParseRange(variable.Versions); err != nil {
-			// 	return fmt.Errorf("template %s has an invalid version range: %s", path, variable.Versions)
-			// }
+			if _, err := semver.ParseRange(variable.Versions); err != nil {
+				return fmt.Errorf("template %s has an invalid version range: %s", path, variable.Versions)
+			}
 
 			allVariables[variable.Name] = variable
 			if variable.Default.ReferenceVar != "" {
 				referenceVarMap[variable.Name] = variable
 			}
+
+			if variable.ConditionalRef.ReferenceVar != "" {
+				conditionRefMap[variable.Name] = variable
+			}
 		}
 
 		for _, currVar := range referenceVarMap {
 			refVar, ok := allVariables[currVar.Default.ReferenceVar]
 			if !ok {
-				return fmt.Errorf("template %s has a variable %s with reference to a non-existent variable: %s", path, currVar.Name, currVar.Default.ReferenceVar)
+				return fmt.Errorf("template %s has a variable %s with default reference to a non-existent variable: %s", path, currVar.Name, currVar.Default.ReferenceVar)
 			}
 
 			if currVar.Name == refVar.Name {
```

Comment:
we could probably remove this check, since `isCyclicalDefaultVariableReference` takes care of this condition

Reply:
Ill keep this top level one as it gives a more descriptive reason as to why the validation is failing but will edit the error message of the cyclical reference to be more correct!